proto/minder/v1/minder.proto

```diff
@@ -1289,6 +1289,20 @@ enum Entity {
     ENTITY_PULL_REQUESTS = 4;
 }
 
+// AutoRegistration is the configuration for auto-registering entities.
+// When nothing is set, it means that auto-registration is disabled. There is no difference between disabled
+// and undefined so for the "let's not auto-register anything" case we'd just let the repeated string empty
+message AutoRegistration {
+    // enabled is the list of entities that are enabled for auto-registration.
+    repeated string enabled = 1;
+}
+
+// ProviderConfig contains the generic configuration for a provider.
+message ProviderConfig {
+    // auto_registration is the configuration for auto-registering entities.
+    AutoRegistration auto_registration = 1;
+}
```

Comment:
I feel like we need some sort of documentation about how these messages line up (or don't) with other fields in the API.  I'm assuming these are serialized as Struct / JSON in the `config` field of `Provider`?

Reply:
Yes, this is the next part I'm working on at the moment, I hope to get a PR out soon.